topgun/k8s/kubernetes_creds_mgmt_test.go

```diff
@@ -71,7 +75,40 @@ var _ = Describe("Kubernetes credential management", func() {
 	})
 
 	Context("Consuming per-team k8s secrets", func() {
+		JustBeforeEach(func() {
```

Comment:
I think we might be able to get rid of this by having these two contexts underneath a third one.

Playing around with it, we could have something like:

```go
	Context("Consuming per-team k8s secrets", func() {
		JustBeforeEach(func() {
			// ((foo)) --> bar
			createCredentialSecret(releaseName, "foo", "main", map[string]string{"value": "bar"})

			// ((caz.baz)) --> zaz
			createCredentialSecret(releaseName, "caz", "main", map[string]string{"baz": "zaz"})

			fly.Run("set-pipeline", "-n", "-c", "../pipelines/minimal-credential-management.yml", "-p", "pipeline")
			session := fly.Start("get-pipeline", "-p", "pipeline")
			Wait(session)

			Expect(string(session.Out.Contents())).ToNot(ContainSubstring("bar"))
			Expect(string(session.Out.Contents())).ToNot(ContainSubstring("zaz"))

			fly.Run("unpause-pipeline", "-p", "pipeline")
		})

		Context("using the default namespace created by the chart", func() {
			It("Gets credentials set by consuming k8s secrets", func() {
				session := fly.Start("trigger-job", "-j", "pipeline/unit", "-w")
				Wait(session)

				Expect(string(session.Out.Contents())).To(ContainSubstring("bar"))
				Expect(string(session.Out.Contents())).To(ContainSubstring("zaz"))
			})
		})

		Context("using a user-provided namespace", func() {
			BeforeEach(func() {
				Run(nil, "kubectl", "create", "namespace", releaseName+"-main")
				extraArgs = []string{
					"--set=concourse.web.kubernetes.createTeamNamespaces=false",
				}
			})

			It("Gets credentials set by consuming k8s secrets", func() {
				session := fly.Start("trigger-job", "-j", "pipeline/unit", "-w")
				Wait(session)

				Expect(string(session.Out.Contents())).To(ContainSubstring("bar"))
				Expect(string(session.Out.Contents())).To(ContainSubstring("zaz"))
			})
		})
	})
```

Wdyt?

Reply:
didn't think a lot about optimization, although it was obvious 😁